libs/dal/src/repositories/notification/notification.repository.ts

```diff
@@ -89,7 +89,17 @@ export class NotificationRepository extends BaseRepository<NotificationEntity> {
             $nin: [StepTypeEnum.TRIGGER],
           },
         },
-        populate: { path: 'executionDetails' },
+        select: 'createdAt digest payload providerId step status type updatedAt',
```

Comment:
This looks awesome 🥳

Reply:
I was hesitant if doing the trimming at controller level with a filtering DTO or here in the query, just in case `getFeedItem` was reused somewhere else. I leave it there for now, but if this is being reused we might think what's the most useful way for us to trim the data.